pkg/statistics/handle/bootstrap.go

```diff
@@ -67,22 +67,31 @@ func (*Handle) initStatsMeta4Chunk(cache statstypes.StatsCache, iter *chunk.Iter
 	for row := iter.Begin(); row != iter.End(); row = iter.Next() {
 		physicalID = row.GetInt64(1)
 		maxPhysicalID = max(physicalID, maxPhysicalID)
+		// During the initialization phase, we need to initialize LastAnalyzeVersion with the snapshot,
+		// which ensures that we don't duplicate the auto-analyze of a particular type of table.
+		// When the predicate columns feature is turned on, if a table has neither predicate columns nor indexes,
+		// then auto-analyze will only analyze the _row_id and refresh stats_meta,
+		// but since we don't have any histograms or topn's created for _row_id at the moment.
+		// So if we don't initialize LastAnalyzeVersion with the snapshot here,
+		// it will stay at 0 and auto-analyze won't be able to detect that the table has been analyzed.
+		// But in the future, we maybe will create some records for _row_id, see:
+		// https://github.com/pingcap/tidb/issues/51098
 		newHistColl := *statistics.NewHistColl(physicalID, row.GetInt64(3), row.GetInt64(2), 4, 4)
 		snapshot := row.GetUint64(4)
+		lastAnalyzeVersion, lastStatsFullUpdateVersion := snapshot, snapshot
+		if !row.IsNull(5) {
+			lastAnalyzeVersion = max(lastAnalyzeVersion, row.GetUint64(5))
+			lastStatsFullUpdateVersion = max(lastStatsFullUpdateVersion, row.GetUint64(5))
+		}
+		if !row.IsNull(6) {
+			lastStatsFullUpdateVersion = max(lastStatsFullUpdateVersion, row.GetUint64(6))
+		}
 		tbl := &statistics.Table{
-			HistColl:              newHistColl,
-			Version:               row.GetUint64(0),
-			ColAndIdxExistenceMap: statistics.NewColAndIndexExistenceMapWithoutSize(),
-			// During the initialization phase, we need to initialize LastAnalyzeVersion with the snapshot,
-			// which ensures that we don't duplicate the auto-analyze of a particular type of table.
-			// When the predicate columns feature is turned on, if a table has neither predicate columns nor indexes,
-			// then auto-analyze will only analyze the _row_id and refresh stats_meta,
-			// but since we don't have any histograms or topn's created for _row_id at the moment.
-			// So if we don't initialize LastAnalyzeVersion with the snapshot here,
-			// it will stay at 0 and auto-analyze won't be able to detect that the table has been analyzed.
-			// But in the future, we maybe will create some records for _row_id, see:
-			// https://github.com/pingcap/tidb/issues/51098
-			LastAnalyzeVersion: snapshot,
+			HistColl:               newHistColl,
+			Version:                row.GetUint64(0),
+			ColAndIdxExistenceMap:  statistics.NewColAndIndexExistenceMapWithoutSize(),
+			LastAnalyzeVersion:     lastAnalyzeVersion,
+			LastStatsUpdateVersion: lastStatsFullUpdateVersion,
```

Comment:
Those names are still quite confusing. At lease, we should unify them to the same name.

Reply:
We have three different names in this PR for the same value.